### PR TITLE
Generate Scalding microsite via sbt-microsites

### PR DESCRIPTION
**Pull request by sritchie**
This PR adds a new Github Pages microsite for Scalding using the [SBT-Microsites plugin](https://github.com/47deg/sbt-microsites).

Here's a preview: <https://sritchie.github.io/scalding>

I've also 

- converted the FAQ over, along with the `HBase and Scalding` page so we have an example for how to convert the rest of the wiki over.
- moved the relevant sidebar links into `Resources for Learners`, and added a `Powered By` page with everyone currently using Scalding.
- added the microsite build to one of the targets in `.travis.yml`
- added gitter build notifications to `.travis.yml`

A huge advantage of this approach over the wiki is that any code snippets we add are going to be checked by the build, thanks to `tut`.

We can also move the tutorials into the microsite and kill those build targets!

### CODE REVIEW

**Comment by sritchie-stripe** [comment]
Confirmed that the docs get built https://travis-ci.org/twitter/scalding/jobs/177091538#L3439

File: docs/src/main/tut/faq.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+Pass the option `--tool.partialok` to your job and it will ignore any missing data. It's safer to work around by either filling with place-holder empty files, or writing sources thatxb will skip known-missing dates. Using that option by default is very dangerous.
+
+### I receive this error when running `sbt update`: <tt>Error occurred during initialization of VM. Incompatible minimum and maximum heap sizes specified</tt>
+
+In your sbt script, set `local min=$(( $mem / 2 ))`
+
+# Writing Jobs
+
+### How do I perform windowed calculations (for example, moving average) in Scalding?
```
Comment (oscar-stripe) [inline docs/src/main/tut/faq.md]: this is pretty old and points to the fields API. I'd nuke it.

File: docs/src/main/tut/faq.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+You want to use `GroupBuilder.scanLeft`.  A `scanLeft` is like a `foldLeft` except that you output each intermediate value.  Both of these functions are part of the standard Scala library as well. [See StackOverflow for `scanLeft` examples](http://stackoverflow.com/questions/6799441/in-scala-how-do-i-keep-track-of-running-totals-without-using-var).  For the specific example of moving averages in Scalding, see the [cascading-user group discussion](https://groups.google.com/forum/?fromgroups#!topic/cascading-user/CTYOjlHs6xE).
+
+### How do I read a single reduced value from a pipe?
```
Comment (oscar-stripe) [inline docs/src/main/tut/faq.md]: this is no longer accurate. There is Execution and ValuePipe, both are related to this problem.

**Comment by sritchie-stripe** [comment]
okay @oscar-stripe, fixed.

**Comment by sritchie-stripe** [comment]
oh no, we're in bizarro stripe world!

**Comment by johnynek** [comment]
👍 merge when green. Can't wait to get this on the site!

File: docs/src/main/tut/resources_for_learners.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Videos
+
+- [Scalding: Powerful & Concise MapReduce Programming](http://www.youtube.com/watch?v=LaAEhPoIm_A)
+- [Scalding lecture for UC Berkeley's Analyzing Big Data with Twitter class](http://blogs.ischool.berkeley.edu/i290-abdt-s12/2012/11/03/video-lecture-intro-to-scalding-by-posco-and-argyris/)
+- [Scalding REPL with Eclipse Scala Worksheets](http://youtu.be/Forl4hpg7kA)
+
+## How-tos
+
+- [[Scalding with CDH3U2 in a Maven project]]
```
Comment (piyushnarang) [inline docs/src/main/tut/resources_for_learners.md]: does this need to be a link?

**Review by piyushnarang — APPROVED** [review]
Thanks for making these changes!

File: docs/src/main/tut/resources_for_learners.md (hunk 3 above)
Comment (sritchie-stripe) [reply docs/src/main/tut/resources_for_learners.md]: yes, good call! added a todo to convert a wiki page over.

File: docs/src/main/tut/index.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+You can find more example code under [examples/](https://github.com/twitter/scalding/tree/master/scalding-commons/src/main/scala/com/twitter/scalding/examples). If you're interested in comparing Scalding to other languages, see our [Rosetta Code page](https://github.com/twitter/scalding/wiki/Rosetta-Code), which has several MapReduce tasks in Scalding and other frameworks (e.g., Pig and Hadoop Streaming).
+
+## Documentation and Getting Started
+
+The latest API docs are hosted at Scalding's [ScalaDoc index](api/).
+
+* [**Getting Started**](https://github.com/twitter/scalding/wiki/Getting-Started) page on the [Scalding Wiki](https://github.com/twitter/scalding/wiki)
+* [Scalding Scaladocs](http://twitter.github.com/scalding) provide details beyond the API References. Prefer using this as it's always up to date.
+* [**REPL in Wonderland**](tutorial/WONDERLAND.md) a hands-on tour of the scalding REPL requiring only git and java installed.
```
Comment (oscar-stripe) [inline docs/src/main/tut/index.md:46]: can we make this a tut compiled thing? Would be awesome to have that checked.

Comment (sritchie-stripe) [reply docs/src/main/tut/index.md:46]: Yes, definitely - next PR!

**Pull request merged**